src/main/java/org/commcare/suite/model/Detail.java

```diff
@@ -321,15 +325,18 @@ public void writeExternal(DataOutputStream out) throws IOException {
             indices.addElement(i);
             continue;
         }
-        if (indices.size() == 0) {
-            return new int[]{};
-        } else {
-            int[] ret = new int[indices.size()];
-            for (int i = 0; i < ret.length; ++i) {
-                ret[i] = indices.elementAt(i);
-            }
-            return ret;
+
+        int resultLength = indices.size() + cacheAndIndexedIndices.size();
```

Comment:
If you have a minute, I'd refactor this into a method (mergeIntegerArrays) simply because it would make the code mildly easier to read.